level1c4pps/fci2pps_lib.py

```diff
@@ -0,0 +1,234 @@
+                "wv_63": "ch_tb67",
+                "ir_133": "ch_tb133",
+                "wv_73": "ch_tb73",
+                "vis_05": "ch_rxx",
+                "vis_04": "ch_rxx",
+                "vis_09": "ch_rxx"}
+
+BANDNAMES = list(PPS_TAGNAMES.keys())
+
+
+def get_encoding_fci(scene):
+    """Get netcdf encoding for all datasets."""
+    return get_encoding(scene,
+                        BANDNAMES,
+                        PPS_TAGNAMES,
+                        chunks=None)
+
+
+def set_header_and_band_attrs(scene, orbit_n=00000):
+    """Set and delete some attributes."""
+    nimg = 0  # name of first dataset is image0
+    # Set some header attributes:
+    irch = scene['ir_105']
+    scene.attrs['source'] = "fci2pps.py"
+    nimg = set_header_and_band_attrs_defaults(scene, BANDNAMES, PPS_TAGNAMES, REFL_BANDS, irch, orbit_n=orbit_n)
+    for band in REFL_BANDS:
+        if band not in scene:
+            continue
+        print("Is this correct???????????????")
+        scene[band].attrs['sun_zenith_angle_correction_applied'] = 'True'
+    return nimg
+
+
+def get_solar_angles(scene, lons, lats):
+    """Compute solar angles.
+
+    Compute angles for each scanline using their acquisition time to account for
+    the earth's rotation over the course of one scan.
+
+    Returns:
+        Solar azimuth angle, Solar zenith angle in degrees
+
+    """
+    suna = np.full(lons.shape, np.nan)
+    sunz = np.full(lons.shape, np.nan)
+    acq_time = scene["ir_105_time"].copy()
+    _, suna = get_alt_az(acq_time, lons, lats)
+    suna = np.rad2deg(suna)
+    sunz = sun_zenith_angle(acq_time, lons, lats)
+    return suna, sunz
+
+
+def fix_time(scene):
+    """Make datetime objects from time in seconds."""
+    if type(scene["ir_105_time"].values[0, 0]) == np.float64:
+        epoch_to_2000 = dt.datetime(2000, 1, 1, tzinfo=dt.timezone.utc).timestamp()
+        scene["ir_105_time"] = scene["ir_105_time"] + epoch_to_2000
+        scene["ir_105_time"] = scene["ir_105_time"].astype('datetime64[s]')
+    ind = np.int16(scene["ir_105_time"].shape[0]/2)
+    a_time = dt64_to_datetime(scene["ir_105_time"].values[ind, ind])
+    if a_time > scene.end_time or a_time < scene.start_time:
+        raise ValueError
+    scene["ir_105_time"].attrs.pop("units", None)
+    scene["ir_105_time"].attrs.pop("_FillValue", None)
+
+def resample_data(scn_in, datasets, resample_grid="coarse"):
+    logger.info("Resampling to coarsest area")
+    if resample_grid in ["fine"]:
+        logger.info("Resampling to finest grid")
+        scn_out = scn_in.resample(scn_in.finest_area(), datsets=datasets, resampler='native')
+    elif resample_grid in ["coarse"]:
+        logger.info("Resampling to coarsest grid")
+        scn_out = scn_in.resample(scn_in.coarsest_area(), datsets=datasets, resampler='native')
+    elif "msg" in resample_grid:
+        import psutil
+        if psutil.virtual_memory()[0]/32e9 < 1:
+            logger.warning(
```

Comment:
I think it would be better with a raise here, and add a choice to perform this type of remapping and advice the user to use this option.